plugins/chat/assets/javascripts/discourse/components/chat-search.gjs

```diff
@@ -0,0 +1,291 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn, hash } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import { service } from "@ember/service";
+import { isBlank } from "@ember/utils";
+import { not } from "truth-helpers";
+import ConditionalLoadingSpinner from "discourse/components/conditional-loading-spinner";
+import DButton from "discourse/components/d-button";
+import DropdownMenu from "discourse/components/dropdown-menu";
+import FilterInput from "discourse/components/filter-input";
+import LoadMore from "discourse/components/load-more";
+import { ajax } from "discourse/lib/ajax";
+import discourseDebounce from "discourse/lib/debounce";
+import { bind } from "discourse/lib/decorators";
+import { INPUT_DELAY } from "discourse/lib/environment";
+import getURL from "discourse/lib/get-url";
+import DiscourseURL from "discourse/lib/url";
+import autoFocus from "discourse/modifiers/auto-focus";
+import tabToSibling from "discourse/modifiers/tab-to-sibling";
+import { i18n } from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+import ChatChannelTitle from "discourse/plugins/chat/discourse/components/chat-channel-title";
+import ChatMessageComponent from "discourse/plugins/chat/discourse/components/chat-message";
+import ChatThreadHeading from "discourse/plugins/chat/discourse/components/chat-thread-heading";
+import ChatChannel from "discourse/plugins/chat/discourse/models/chat-channel";
+import ChatMessage from "discourse/plugins/chat/discourse/models/chat-message";
+
+export default class ChatSearch extends Component {
+  @service router;
+
+  @tracked currentSort = "relevance";
+  @tracked messages = [];
+  @tracked isLoading = false;
+  @tracked offset = 0;
+  @tracked hasMoreResults = false;
+  @tracked lastQuery = null;
+
+  limit = 20;
```

Comment:
Probably doesn't make too much difference, but don't we usually put these in a const above the component .e.g 

```javascript
const LIMIT = 20;
```